src/tiledb/cloud/vcf/ingestion.py

```diff
@@ -0,0 +1,987 @@
+    :param id: profiler event id, defaults to "consolidate"
+    """
+
+    logger, profiler = setup(config, dataset_uri, id)
+
+    if type(exclude) == str:
+        exclude = [exclude]
+
+    if type(include) == str:
+        include = [include]
+
+    group = tiledb.Group(dataset_uri)
+
+    for member in group:
+        uri = member.uri
+        name = member.name
+
+        # Skip excluded and non-included arrays
+        if (exclude and name in exclude) or (include and name not in include):
+            continue
+
+        # NOTE: REST currently only supports fragment_meta, commits, metadata
+        modes = ["commits", "fragment_meta", "array_meta"]
+
+        # Consolidate fragments for log and manifest arrays
+        if name in [LOG_ARRAY, MANIFEST_ARRAY]:
+            modes += ["fragments"]
+
+        for mode in modes:
+            config = tiledb.Config({"sm.consolidation.mode": mode})
+            try:
+                tiledb.consolidate(uri, config=config)
+            except Exception as e:
+                print(e)
+
+        for mode in modes:
+            config = tiledb.Config({"sm.vacuum.mode": mode})
+            try:
+                tiledb.vacuum(uri, config=config)
+            except Exception as e:
+                print(e)
+
+    profiler.finish()
+
+
+# --------------------------------------------------------------------
+# DAGs
+# --------------------------------------------------------------------
+
+
+def ingest_manifest_dag(
+    dataset_uri: str,
+    *,
+    config: Optional[Mapping[str, Any]] = None,
+    namespace: Optional[str] = None,
+    search_uri: Optional[str] = None,
+    pattern: Optional[str] = None,
+    ignore: Optional[str] = None,
+    sample_list_uri: Optional[str] = None,
+    max_files: Optional[int] = None,
+    batch_size: int = MANIFEST_BATCH_SIZE,
+    workers: int = MANIFEST_WORKERS,
+    extra_attrs: Optional[Union[Sequence[str], str]] = None,
+    vcf_attrs: Optional[str] = None,
+):
+    """
+    Create a DAG to load the manifest array.
+
+    :param dataset_uri: dataset URI
+    :param config: config dictionary, defaults to None
+    :param namespace: TileDB-Cloud namespace, defaults to None
+    :param search_uri: URI to search for VCF files, defaults to None
+    :param pattern: pattern to match when searching for VCF files, defaults to None
+    :param ignore: pattern to ignore when searching for VCF files, defaults to None
+    :param sample_list_uri: URI with a list of VCF URIs, defaults to None
+    :param max_files: maximum number of URIs to ingest, defaults to None
+    :param batch_size: manifest batch size, defaults to MANIFEST_BATCH_SIZE
+    :param workers: maximum number of parallel workers, defaults to MANIFEST_WORKERS
+    :param extra_attrs: INFO/FORMAT fields to materialize, defaults to None
+    :param vcf_attrs: VCF with all INFO/FORMAT fields to materialize, defaults to None
+    """
+
+    logger = get_logger()
+
+    graph = tiledb.cloud.dag.DAG(
```

Comment:
This is where I think we should be using a dynamic dag instead of running three separate dags. @NikolaosPapailiou Can you work with @gspowley and show this could be switched to a dynamic data for determining the files to ingest.

Reply:
Sure I will be happy to help with this. The main idea would be to have only one DAG that does all the processing (generate_uris, update_manifest, ingest). Dynamic taskgraphs can help with dynamically creating tasks based on the output of other task. So in this example generate_uris could return a list of sample uris that update_manifest and ingest tasks expand upon creating one task per sample uri. This can be done by just returning a list object and adding `expand_output` on the dynamic DAG node (you can check an example in https://github.com/TileDB-Inc/TileDB-Cloud-Py/blob/main/tests/test_dag.py#L643).

Please keep in mind that the `expand_output` functionality is only available for bath taskgraphs. So if you want the same logic to run with realtime taskgraphs the code needs to be duplicated and follow another pattern for local or realtime execution.